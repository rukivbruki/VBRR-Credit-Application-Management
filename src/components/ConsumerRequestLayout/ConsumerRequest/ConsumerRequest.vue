<template>
  <div class="el-consumer-request-container">
    <el-divider content-position="left">
      <h2 class="el-consumer-request-header">Идентификация персоны</h2>
    </el-divider>
    <div class="el-consumer-request-warning">
      <p class="el-consumer-request-warning-text">
        ВАЖНО! Перед поиском клиента проверить персональные данные в Террасофте,
        передать данные в Афину. Исправления в Егаре заблокированы.
      </p>
    </div>
    <PersonalDataForms />
    <TableWithPersonalData
      :isSelectBtnClicked="isSelectBtnClicked"
      @select-person-id="selectedPersonId = $event"
    />
    <router-link
      class="el-link-inBtn"
      :to="`/consumer-request/${selectedPersonId}/info`"
    >
      <el-button v-if="getCroppedPersonList.length > 0" @click="selectBtnClick"
        >Выбрать
      </el-button>
    </router-link>
  </div>
</template>
<script>
import PersonalDataForms from '../PersonalDataForms/PersonalDataForms';
import TableWithPersonalData from '../TableWithPersonalData/TableWithPersonalData';
import { mapGetters } from 'vuex';

export default {
  name: 'ConsumerRequest',
  components: { PersonalDataForms, TableWithPersonalData },
  data() {
    return {
      isSelectBtnClicked: false,
      selectedPersonId: null,
    };
  },
  computed: mapGetters(['getCroppedPersonList']),
  methods: {
    selectBtnClick() {
      this.isSelectBtnClicked = true;
      setTimeout(() => (this.isSelectBtnClicked = false), 1000);
    },
  },
};
</script>

<style scoped>
.el-consumer-request-container {
  margin: 35px;
}

.el-consumer-request-header {
  font-style: normal;
  font-size: 20px;
  color: #333333;
}

.el-consumer-request-warning {
  display: none;
  background-color: rgb(255, 211, 0);
  margin-top: 5px;
  padding: 4px 4px 4px 20px;
}

.el-consumer-request-warning-text {
  font-family: Roboto, sans-serif;
  font-style: normal;
  font-weight: normal;
}

.el-button {
  background-color: rgb(255, 211, 0);
  margin-bottom: 20px;
}

.el-link-inBtn {
  text-decoration: none;
  color: #606266;
}
</style>
