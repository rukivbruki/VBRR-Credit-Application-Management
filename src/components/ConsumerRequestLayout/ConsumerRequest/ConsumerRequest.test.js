import { mount } from '@vue/test-utils';
import ConsumerRequest from './ConsumerRequest';
describe('ConsumerRequest.vue render', () => {
  test('renders correctly', () => {
    expect(mount(ConsumerRequest).element).toMatchSnapshot();
  });
});
