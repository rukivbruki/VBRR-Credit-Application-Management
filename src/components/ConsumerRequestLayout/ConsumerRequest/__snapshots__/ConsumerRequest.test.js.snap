// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerRequest.vue render renders correctly 1`] = `
<div>
  <h2
    class="el-ConsumerRequest-header"
  >
    Идентификация персоны
  </h2>
  <div
    class="el-ConsumerRequest-warning"
  >
    <p
      class="el-ConsumerRequest-warning-text"
    >
       ВАЖНО! Перед поиском клиента проверить персональные данные в Террасофте, передать данные в Афину. Исправления в Егаре заблокированы. 
    </p>
  </div>
  <el-form
    label-width="120px"
    model="[object Object]"
    style="padding-top: 20px;"
  >
    <el-form-item
      label="Имя"
    >
      <el-input
        class="el-form-input"
      />
    </el-form-item>
    <el-form-item
      label="Фамилия"
    >
      <el-input
        class="el-form-input"
      />
    </el-form-item>
    <el-form-item
      label="Отчество"
    >
      <el-input
        class="el-form-input"
      />
    </el-form-item>
    <el-form-item
      label="Дата рождения"
    >
      <el-col
        span="11"
      >
        <el-date-picker
          placeholder="Введите дату"
          style="width: 100%;"
          type="date"
        />
      </el-col>
    </el-form-item>
    <el-form-item>
      <el-button
        style="background-color: rgb(255, 211, 0);"
      >
        Найти 
      </el-button>
    </el-form-item>
  </el-form>
  
  <el-table
    data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    highlight-current-row=""
    style="width: 100%;"
  >
    <el-table-column
      type="index"
      width="50"
    />
    <el-table-column
      label="ФИО"
      property="name"
    />
    <el-table-column
      label="Дата рождения"
      property="date"
    />
    <el-table-column
      label="Тип документа"
      property="typeOfDocument"
    />
    <el-table-column
      label="Серия документа"
      property="documentSeries"
    />
    <el-table-column
      label="Номер документа"
      property="documentNumber"
    />
    <el-table-column
      label="Адресс прописки"
      property="address"
    />
  </el-table>
  <div
    class="block"
  >
    <el-pagination
      background=""
      layout="prev, pager, next"
      total="20"
    />
  </div>
  
  <el-form-item
    style="padding-left: 60px;"
  >
    <el-button
      style="background-color: rgb(255, 211, 0);"
    >
      Выбрать 
    </el-button>
  </el-form-item>
</div>
`;
